File: practices/VW05p_Enrichement.cpp
//author@nowadays - 2021-05-08  10:00:52
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false), cin.tie(NULL), cout.tie(NULL);
    int n, m;
    cin >> n >> m;
    vector<vector<int>> a(n + 1, vector<int>(m + 1));
    for (int i = 1; i <= n; ++i) {
        for (int j = 1; j <= m; ++j) {
            cin >> a[i][j];
        }
    }
    assert(3 <= n && 3 <= m && n * m <= 1000000);
    int ans = numeric_limits<int>::max();
    for (int i = 1; i <= n - 2; ++i) {
        for (int j = 1; j <= m - 2; ++j) {
            int tmp = 0;
            for (int dx = 0; dx < 3; ++dx) {
                for (int dy = 0; dy < 3; ++dy) {
                    tmp += a[i + dx][j + dy];
                }
            } 
            ans = min(ans, tmp);
        }
    }
    cout << ans << "\n";
    return 0;
}

File: practices/VW05p_Enrichement.py
n, m = [int(i) for i in input().strip().split(' ')]
a = [None] * n 
for i in range(n):
    a[i] = [int(x) for x in input().strip().split(' ')]
ans = int(1e18)
for i in range(n - 2):
    for j in range(m - 2):
        tmp = 0
        for dx in range(3):
            for dy in range(3):
                tmp += a[i + dx][j + dy]
        ans = min(ans, tmp) 
print(ans)  
